### DISCUSSION TITLE
Is ClickHouse really mandatory here?

### DISCUSSION THREAD

**Discussion by naddika** [plausible/analytics#2179] [Self-Hosted Support]
Is ClickHouse really mandatory here?
If it wasn't in place, would everything still work well? Or would it work well up to a certain workload? What would it be approximately?

I'm thinking of turning it off because in my case it may not be needed. However, some things are  too intertwined with it.
Is there a simple way to turn if off so that, obviously, everything else would continue to fuction properly.

**Comment by ruslandoga** [comment]
@naddika 👋

Events (like pageviews) and session data is stored in ClickHouse so prior to turning it off you'd need to modify the logic to store this data somewhere else. And then the queries to read events and sessions use ClickHouse-specific functions, so those would also need to be changed to adapt to your new storage.
